core/src/data/value/binary_op/decimal.rs

```diff
@@ -0,0 +1,257 @@
+use {
+    super::TryBinaryOperator,
+    crate::{data::ValueError, prelude::Value, result::Result},
+    rust_decimal::prelude::Decimal as Dec,
+    std::cmp::Ordering,
+    Value::*,
+};
+
+impl PartialEq<Value> for Dec {
+    fn eq(&self, other: &Value) -> bool {
+        match other {
+            I8(other) => *self == Dec::from(*other),
+            I64(other) => *self == Dec::from(*other),
+            F64(other) => *self == Dec::from_f64_retain(*other).unwrap(),
+            Decimal(other) => *self == *other,
+            _ => false,
+        }
+    }
+}
+
+impl PartialOrd<Value> for Dec {
+    fn partial_cmp(&self, other: &Value) -> Option<Ordering> {
+        match *other {
+            I8(rhs) => self.partial_cmp(&(Dec::from(rhs))),
+            I64(rhs) => self.partial_cmp(&(Dec::from(rhs))),
+            F64(rhs) => self.partial_cmp(&(Dec::from_f64_retain(rhs).unwrap())),
```

Comment:
`PartialOrd` is thinking of returning an Option, so it seems better to return None if None is found.

Reply:
+1